File: fsharp/darts/Darts.fs
module Darts

let score (x: double) (y: double): int =
    let hypotSq = x * x + y * y
    match hypotSq with
    | hypotSq when hypotSq <= 1.0 -> 10
    | hypotSq when hypotSq <= 25.0 -> 5
    | hypotSq when hypotSq <= 100.0 -> 1
    | _ -> 0

File: csharp/darts/Darts.cs
public static class Darts
{
    public static int Score(double x, double y)
    {
        int result = 0;
        double distSq = x * x + y * y;
        if (distSq <= 1)
        {
            result = 10;
        }
        else if (distSq <= 25)
        {
            result = 5;
        }
        else if (distSq <= 100)
        {
            result = 1;
        }
        return result;
    }
}
